numojo/__init__.mojo

```diff
@@ -12,3 +12,9 @@ alias __version__ = "V0.2"
 alias pi = core.constants.Constants.pi
 alias e = core.constants.Constants.e
 alias c = core.constants.Constants.c
+
+# core alias
+alias idx = Idx
+alias ndarray = NDArray
+alias ndshape = NDArrayShape
+alias ndstride = NDArrayStride
```

Comment:
I think it's good for users since it's closer to numpy and easy to type. For example
```
var A = nm.NDArray[i64](3, 4, 5) # random capital letters makes it a bit awkward. 
var A = nm.ndarray[i64](3, 4, 5) # This is easier to type and is closer to np.ndarray. 
```
The above is the reason I did the alias. we could change the `ndarray.mojo` to `array.mojo` or something similar in case it clashes with any definitions.

Reply:
It is easier to type, but not necessarily closer to numpy in several aspects:

1. In Python, classes should be named with CamelCase (PEP8), only primitive types can be named with lower cases (I think that `numpy.ndarray` is not a proper name, should be `numpy.NDArray`). In Mojo, struct names should be CamelCase. Even though we want to mimic numpy, we should be strictly consistent with Mojo's naming convention and avoid confusion.

2. In numpy, `numpy.ndarray` is a class, and it only acts as a container. Numpy does not want users to directly use this class for initializing an `ndarray`, but better via the function `numpy.array()`. Thus, `numpy.ndarray(1,2,3)` is not a legal expression.
 
Thus, in Numojo, `var A = nm.ndarray[i64](3, 4, 5)` would not be a preferred way for users to create a concrete `ndarray` too. Users should do this via a function 
```mojo
var A = nm.array[i64](shape=Shape(3,4,5))
```

5. Moreover, `numpy.ndarray` does not take in a series of integers as the shape, but it take in a tuple as shape. This is something we can change in future.

Above all, we should make `NDArray` an abstract container. To create a concreate array, we should use the function `numojo.array()` instead.

Reply:
Another issue is that the name `ndarray` would conflict with the `ndarray` module.

Reply:
Right, We could then remove the NDArray, Shape, Stride alias and define an `alias array = NDArray` to have the option of initialising arrays from `nm.array(...)`. How about that?

Reply:
We already have the function `array()` in the array creation routine. We can simply add more reloading to that.

Reply:
#110 also discussed about this numojo.array() function.

Reply:
Instead of adding multiple `nm.array` overloading functions, wouldn't it be more easier to define `alias array = NDArray` since both does the same job but the latter with less lines.

Reply:
> Instead of adding multiple `nm.array` overloading functions, wouldn't it be more easier to define `alias array = NDArray` since both does the same job but the latter with less lines.

I do not think that is a proper way. Function and Struct are different things. We should not abuse the `__init__` function of a struct, but try to keep the API clean. The struct `NDArray` is only a container with necessary methods. In most of time, it is not directly exposed to users. The function `array` is a way to create an instance of NDArray.

By this separation, we can also better modularize our package and avoid conflicts.

Reply:
I see. alright! I will make the changes.